doozerlib/cli/images_upstreampulls.py

```diff
@@ -0,0 +1,62 @@
+# stdlib
+import os
+import yaml
+
+# external
+from github import Github
+
+# doozerlib
+from doozerlib import constants
+from doozerlib.cli import cli, pass_runtime
+
+github_token = os.getenv(constants.GIT_TOKEN)
+github_client = Github(github_token)
+repo = github_client.get_repo("openshift/ocp-build-data")
+
+
+@cli.command("images:upstreampulls", short_help="Check upstream PRs in open status")
+@pass_runtime
+def images_upstreampulls(runtime):
+    runtime.initialize(clone_distgits=False, clone_source=False)
+    version = runtime.group.split("-")[-1]
+    retdata = {}
+    contents = repo.get_contents("images", ref="openshift-" + version)
+    for content_file in contents:
+        ydata = yaml.load(content_file.decoded_content, Loader=yaml.FullLoader)
+        if 'content' not in ydata:
+            continue
+        if 'git' not in ydata['content']['source']:
+            continue
+        upstream_project = ydata['content']['source']['git']['url'].split(':')[-1].split('/')[0]
+        upstream = ydata['content']['source']['git']['url'].split('/')[-1].split('.')[0]
+        if upstream_project == "openshift-priv":
+            xrepo = github_client.get_repo("openshift/" + upstream)
+        else:
+            xrepo = github_client.get_repo(upstream_project + "/" + upstream)
```

Comment:
See https://github.com/openshift/doozer/blob/c7e31761b3ec312d4c3101bd3c93c94fc53c5025/doozerlib/cli/images_streams.py#L650 for how to correctly resolve the upstream repo url and branch for each image.